libclc/ptx-nvidiacl/libspirv/SOURCES

```diff
@@ -71,6 +71,8 @@ relational/isfinite.cl
 relational/isinf.cl
 relational/isnan.cl
 synchronization/barrier.cl
+async/async_work_group_strided_copy.cl
+async/wait_group_events.cl
```

Comment:
I think libclc/ptx-nvidiacl/include/clc/async/gentype.inc must be added here too. 
Otherwise, libclc build rules won't re-build the library when libclc/ptx-nvidiacl/include/clc/async/gentype.inc changes.

Please, add all files from include directory.

I would also check that *.inc, *.h files added for CUDA/HIP backends are included in corresponding SOURCES files.

Reply:
OK I'll have to try to understand this build system better.  There is e.g. ptx-nvidiacl/libspirv/math/fma.inc that is not added to sources, although it is included in a directory with the corresponding source file, fma.cl.  Should I add all such .inc files to SOURCES?
The libclc/generic/include/clc/async/gentype.inc is also not added to the generic/libspirv/SOURCES file.  Is this a mistake or is it only necessary to add libclc/ptx-nvidiacl/include/clc/async/gentype.inc to ptx-nvidiacl/libspirv/SOURCES?

Reply:
I might be wrong as I'm not an expert in libclc - it's not used for Intel targets.
I think we can conduct an experiment to check if dependencies are properly handled by the build system.
Could you build libclc, add an empty space to fma.inc, do incremental build for libclc again and check that only fma.cl is built, please?
I'm not sure if clang is able to communicate dependencies to the build system automatically based on preprocessor output for OpenCL language mode.

Reply:
So CMake is setting CMAKE_CONFIGURE_DEPENDS for all files listed in the SOURCES file, then all source files listed in SOURCES are built incrementally.  However if you add a .inc file to SOURCES that .inc file is not built incrementally: if you also change the .cl source file that includes the .inc file both the .cl file and the .inc file are built incrementally, irrespective of whether the .inc file was included in SOURCES or not.  

I don't have an immediate fix to make .inc files build incrementally without also changing the .cl file that they are included from, but I could make an issue for this?

Reply:
I don't have a strong opinion, but comments like this: https://github.com/intel/llvm/issues/5429#issuecomment-1049413150 makes me think that there might be some problems with moving libclc functionality from *.cl files to *.inc or *.h. 
I think it mostly impacts developers building libclc incrementally and I hope it doesn't impact our CI, where we use ccache.

> I don't have an immediate fix to make .inc files build incrementally without also changing the .cl file that they are included from, but I could make an issue for this?

I'll leave it for @AerialMantis.

Reply:
I'm merging the PR as the issue we discuss in this conversion is not blocking.

Reply:
I think it's worth creating a ticket for this, there are quite a few places where `.inc` and `.h` files are used and this is a bit of a pain point for working with libclc. I'm not a CMake expert but I believe I have seen this problem solved before, perhaps by also adding the `.inc` and `.h` files with `target_include_directories`, though that may require more complex parsing of the `SOURCES` files to distinguish those from `.cl` files.